### PR TITLE
feat: add samples for http cors

### PR DESCRIPTION
**Pull request by yuryu**
Add Functions samples and tests for HTTP CORS requests. This is based on the Python version here: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/1523c225344bd7c38ab3b54e7bccf8ef9e7a3184/functions/http/main.py#L127-L152

### CODE REVIEW

File: functions/http/cors.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "functions_framework"
+
+# [START functions_http_cors]
+FunctionsFramework.http "cors_enabled_function" do |_request|
```
Comment (dazuma) [inline functions/http/cors.rb:18]: Use `request` without an underscore. (Convention is for leading underscore on unused parameters only.) Same for the other function below.

Comment (yuryu) [reply functions/http/cors.rb]: Changed!

**Comment by yuryu** [comment]
Looks like the failed tests are unrelated to the change?

**Comment by dazuma** [comment]
Yeah the failure is unrelated and will be addressed by #617.

**Pull request merged**